### PR TITLE
[NO QA] migrate group 5 tests to ts

### PR DESCRIPTION
**Pull request by pasyukevich**
### Fixed Issues

$ https://github.com/Expensify/App/issues/25327 
https://github.com/Expensify/App/issues/25326 
https://github.com/Expensify/App/issues/25325  
https://github.com/Expensify/App/issues/25328
PROPOSAL:

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: tests/actions/ReportTest.ts [hunk 1]
```diff
@@ -21,7 +22,7 @@ import waitForNetworkPromises from '../utils/waitForNetworkPromises';
 
 const UTC = 'UTC';
 jest.mock('../../src/libs/actions/Report', () => {
-    const originalModule = jest.requireActual('../../src/libs/actions/Report');
+    const originalModule: typeof Report = jest.requireActual('../../src/libs/actions/Report');
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:25]:
```suggestion
    const originalModule = jest.requireActual<typeof Report>('@libs/actions/Report');
```

File: tests/actions/ReportTest.ts
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:1]: What about converting all import paths to aliases e.g. `@libs/...`?

File: tests/unit/APITest.ts
Comment (fabioh8010) [inline tests/unit/APITest.ts:1]: What about converting all import paths to aliases e.g. `@src/libs/...`?

File: tests/unit/MigrationTest.ts
Comment (fabioh8010) [inline tests/unit/MigrationTest.ts:1]: What about converting all import paths to aliases e.g. `@src/libs/...`?

File: tests/unit/NetworkTest.ts
Comment (fabioh8010) [inline tests/unit/NetworkTest.ts:1]: What about converting all import paths to aliases e.g. `@src/libs/...`?

File: tests/actions/ReportTest.ts [hunk 2]
```diff
@@ -35,7 +36,7 @@ describe('actions/Report', () => {
         PusherHelper.setup();
         Onyx.init({
             keys: ONYXKEYS,
-            registerStorageEventListener: () => {},
+            //   registerStorageEventListener: () => {},
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:39]:
```suggestion
```

You can remove since `registerStorageEventListener` doesn't exist on Onyx.

File: tests/actions/ReportTest.ts [hunk 3]
```diff
@@ -52,12 +53,12 @@ describe('actions/Report', () => {
     afterEach(PusherHelper.teardown);
 
     it('should store a new report action in Onyx when onyxApiUpdate event is handled via Pusher', () => {
-        global.fetch = TestHelper.getGlobalFetchMock();
+        global.fetch = TestHelper.getGlobalFetchMock() as typeof fetch;
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:56]:
```suggestion
        // @ts-expect-error TODO: Remove this once TestHelper (https://github.com/Expensify/App/issues/25318) is migrated to TypeScript.
        global.fetch = TestHelper.getGlobalFetchMock();
```

I think it's best to just suppress the error until that file is migrated.

File: tests/actions/ReportTest.ts [hunk 4]
```diff
@@ -88,7 +89,7 @@ describe('actions/Report', () => {
                 return waitForBatchedUpdates();
             })
             .then(() => {
-                const resultAction = _.first(_.values(reportActions));
+                const resultAction: OnyxEntry<OnyxTypes.ReportAction> = Object.values(reportActions ?? [])[0];
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:92]:
```suggestion
                const resultAction: OnyxEntry<OnyxTypes.ReportAction> = Object.values(reportActions ?? {})[0];
```

File: tests/actions/ReportTest.ts [hunk 5]
```diff
@@ -167,7 +168,7 @@ describe('actions/Report', () => {
         return TestHelper.signInWithTestUser(TEST_USER_ACCOUNT_ID, TEST_USER_LOGIN)
             .then(() => TestHelper.setPersonalDetails(TEST_USER_LOGIN, TEST_USER_ACCOUNT_ID))
             .then(() => {
-                global.fetch = TestHelper.getGlobalFetchMock();
+                global.fetch = TestHelper.getGlobalFetchMock() as typeof fetch;
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:171]: Same

File: tests/actions/ReportTest.ts [hunk 6]
```diff
@@ -186,27 +187,27 @@ describe('actions/Report', () => {
...
-                const addCommentCalls = _.filter(global.fetch.mock.calls, (callArguments) => callArguments[URL_ARGUMENT_INDEX].includes('AddComment'));
+                const addCommentCalls = (global.fetch as jest.Mock).mock.calls.filter((callArguments) => callArguments[URL_ARGUMENT_INDEX].includes('AddComment'));
                 expect(addCommentCalls.length).toBe(1);
             });
     });
 
     it('should be updated correctly when new comments are added, deleted or marked as unread', () => {
         jest.useFakeTimers();
-        global.fetch = TestHelper.getGlobalFetchMock();
+        global.fetch = TestHelper.getGlobalFetchMock() as typeof fetch;
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:197]: Same

File: tests/actions/ReportTest.ts [hunk 7]
```diff
@@ -424,7 +427,7 @@ describe('actions/Report', () => {
          * already in the comment and the user deleted it on purpose.
          */
 
-        global.fetch = TestHelper.getGlobalFetchMock();
+        global.fetch = TestHelper.getGlobalFetchMock() as typeof fetch;
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:430]: Same

File: tests/actions/ReportTest.ts [hunk 8]
```diff
@@ -536,11 +539,11 @@ describe('actions/Report', () => {
     });
 
     it('should properly toggle reactions on a message', () => {
-        global.fetch = TestHelper.getGlobalFetchMock();
+        global.fetch = TestHelper.getGlobalFetchMock() as typeof fetch;
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:542]: Same

File: tests/actions/ReportTest.ts [hunk 9]
```diff
@@ -650,38 +653,38 @@ describe('actions/Report', () => {
...
                         const reportActionReaction = reportActionsReactions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_REACTIONS}${reportActionID}`];
-                        expect(reportActionReaction[EMOJI.name].users[TEST_USER_ACCOUNT_ID]).toBeUndefined();
+                        expect(reportActionReaction?.[EMOJI.name].users[TEST_USER_ACCOUNT_ID]).toBeUndefined();
                     });
             });
     });
 
     it("shouldn't add the same reaction twice when changing preferred skin color and reaction doesn't support skin colors", () => {
-        global.fetch = TestHelper.getGlobalFetchMock();
+        global.fetch = TestHelper.getGlobalFetchMock() as typeof fetch;
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:662]: Same

File: tests/unit/APITest.ts [hunk 10]
```diff
@@ -1,5 +1,6 @@
-import Onyx from 'react-native-onyx';
-import _ from 'underscore';
+// import Onyx from 'react-native-onyx';
```
Comment (fabioh8010) [inline tests/unit/APITest.ts:1]:
```suggestion
import MockedOnyx from 'react-native-onyx';
```

File: tests/unit/APITest.ts [hunk 11]
```diff
@@ -1,5 +1,6 @@
...
+import type {ValueOf} from 'type-fest';
+import reactNativeOnyxMock from '../../__mocks__/react-native-onyx';
```
Comment (fabioh8010) [inline tests/unit/APITest.ts:3]:
```suggestion
import type {ReactNativeOnyxMock} from '../../__mocks__/react-native-onyx';
```

Export the `ReactNativeOnyxMock` in that file.

File: tests/unit/APITest.ts [hunk 12]
```diff
@@ -14,16 +15,26 @@ import * as TestHelper from '../utils/TestHelper';
 import waitForBatchedUpdates from '../utils/waitForBatchedUpdates';
 import waitForNetworkPromises from '../utils/waitForNetworkPromises';
 
+const Onyx = reactNativeOnyxMock;
```
Comment (fabioh8010) [inline tests/unit/APITest.ts:18]:
```suggestion
const Onyx = MockedOnyx as ReactNativeOnyxMock;
```

Use it here and that should be enough

File: tests/unit/APITest.ts [hunk 13]
```diff
@@ -14,16 +15,26 @@ import * as TestHelper from '../utils/TestHelper';
...
+    jsonCode?: ValueOf<typeof CONST.JSON_CODE>;
+    title?: ValueOf<typeof CONST.ERROR_TITLE>;
+    type?: ValueOf<typeof CONST.ERROR_TYPE>;
+};
+
 const originalXHR = HttpUtils.xhr;
 
 beforeEach(() => {
-    global.fetch = TestHelper.getGlobalFetchMock();
+    global.fetch = TestHelper.getGlobalFetchMock() as typeof fetch;
```
Comment (fabioh8010) [inline tests/unit/APITest.ts:37]: Same

File: tests/unit/APITest.ts [hunk 14]
```diff
@@ -120,8 +136,11 @@ describe('APITests', () => {
 
     test('Write request should not be cleared until a backend response occurs', () => {
         // We're setting up xhr handler that will resolve calls programmatically
-        const xhrCalls = [];
-        const promises = [];
+        const xhrCalls: Array<{
+            resolve: (value: Response | PromiseLike<Response>) => void;
+            reject: (value: unknown) => void;
+        }> = [];
```
Comment (fabioh8010) [inline tests/unit/APITest.ts:142]: Extract `{...}` to a separate type

File: tests/unit/APITest.ts [hunk 15]
```diff
@@ -177,32 +198,30 @@ describe('APITests', () => {
...
                 })
         );
     });
 
     // Given a retry response create a mock and run some expectations for retrying requests
-    const retryExpectations = (retryResponse) => {
-        let successfulResponse = {
+
+    const retryExpectations = (Response: Response) => {
+        const successfulResponse = {
```
Comment (fabioh8010) [inline tests/unit/APITest.ts:209]:
```suggestion
        const successfulResponse: Response = {
```

File: tests/unit/APITest.ts [hunk 16]
```diff
@@ -177,32 +198,30 @@ describe('APITests', () => {
...
+                    expect(PersistedRequests.getAll().length).toEqual(0);
                 })
         );
     });
 
     // Given a retry response create a mock and run some expectations for retrying requests
-    const retryExpectations = (retryResponse) => {
-        let successfulResponse = {
+
+    const retryExpectations = (Response: Response) => {
```
Comment (fabioh8010) [inline tests/unit/APITest.ts:208]:
```suggestion
    const retryExpectations = (response: Response) => {
```

Comment (fabioh8010) [reply tests/unit/APITest.ts:208]: Or let's go back to `retryResponse`

File: tests/unit/APITest.ts [hunk 17]
```diff
@@ -258,7 +277,7 @@ describe('APITests', () => {
         // Given the response data returned when auth is down
         const responseData = {
```
Comment (fabioh8010) [inline tests/unit/APITest.ts:278]:
```suggestion
        const responseData: Response = {
```

File: tests/unit/MigrationTest.ts [hunk 18]
```diff
@@ -7,13 +8,13 @@ import waitForBatchedUpdates from '../utils/waitForBatchedUpdates';
 
 jest.mock('../../src/libs/getPlatform');
 
-let LogSpy;
+let LogSpy: unknown;
```
Comment (fabioh8010) [inline tests/unit/MigrationTest.ts:11]:
```suggestion
let LogSpy: jest.SpyInstance<void, Parameters<typeof Log['info']>>;
```

File: tests/unit/MigrationTest.ts [hunk 19]
```diff
@@ -87,12 +90,13 @@ describe('Migrations', () => {
...
                             };
-                            expect(allReportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1`]).toMatchObject(expectedReportAction);
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1`]).toMatchObject(expectedReportAction);
                         },
                     });
                 }));
 
         it('Should skip zombie report actions and proceed to remove all reportActions given that a previousReportActionID does not exist', () =>
+            // @ts-expect-error Preset necessary values
             Onyx.multiSet({
```
Comment (fabioh8010) [inline tests/unit/MigrationTest.ts:100]: Please have a look [at this discussion on Slack](https://callstack-hq.slack.com/archives/C05LWS6EL3E/p1709223334828259).

File: tests/unit/MigrationTest.ts [hunk 20]
```diff
@@ -117,15 +121,16 @@ describe('Migrations', () => {
...
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1`]).toMatchObject(expectedReportAction);
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}2`]).toBeUndefined();
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}3`]).toBeUndefined();
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}4`]).toMatchObject(expectedReportAction);
                         },
                     });
                 }));
 
         it('Should skip zombie report actions and should not remove any report action given that previousReportActionID exists in first valid report action', () =>
+            // @ts-expect-error Preset necessary values
```
Comment (fabioh8010) [inline tests/unit/MigrationTest.ts:133]: Same

File: tests/unit/MigrationTest.ts [hunk 21]
```diff
@@ -160,15 +165,16 @@ describe('Migrations', () => {
...
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1`]).toMatchObject(expectedReportAction1);
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}2`]).toBeUndefined();
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}3`]).toBeUndefined();
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}4`]).toMatchObject(expectedReportAction4);
                         },
                     });
                 }));
 
         it('Should skip if no valid reportActions', () =>
+            // @ts-expect-error Preset necessary values
```
Comment (fabioh8010) [inline tests/unit/MigrationTest.ts:177]: Same

File: tests/unit/MigrationTest.ts [hunk 22]
```diff
@@ -200,6 +206,7 @@ describe('Migrations', () => {
             ));
 
         it('Should move individual draft to a draft collection of report', () =>
+            // @ts-expect-error Preset necessary values
```
Comment (fabioh8010) [inline tests/unit/MigrationTest.ts:209]: Same

File: tests/unit/MigrationTest.ts [hunk 23]
```diff
@@ -221,16 +228,17 @@ describe('Migrations', () => {
...
+                            expect(allReportActionsDrafts?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}1_2`]).toBeUndefined();
+                            expect(allReportActionsDrafts?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}2_4`]).toBeUndefined();
+                            expect(allReportActionsDrafts?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}1`]).toMatchObject(expectedReportActionDraft1);
+                            expect(allReportActionsDrafts?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}2`]).toMatchObject(expectedReportActionDraft2);
                         },
                     });
                 }));
 
         it('Should skip if nothing to migrate', () =>
+            // @ts-expect-error Preset necessary values
```
Comment (fabioh8010) [inline tests/unit/MigrationTest.ts:241]: Same

File: tests/unit/MigrationTest.ts [hunk 24]
```diff
@@ -246,15 +254,16 @@ describe('Migrations', () => {
...
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}1_1`]).toBeUndefined();
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}1_2`]).toBeUndefined();
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}2_4`]).toBeUndefined();
+                            expect(allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}2`]).toMatchObject(expectedReportActionDraft);
                         },
                     });
                 }));
 
         it("Shouldn't move empty individual draft to a draft collection of report", () =>
+            // @ts-expect-error Preset necessary values
```
Comment (fabioh8010) [inline tests/unit/MigrationTest.ts:266]: Same

File: tests/unit/NetworkTest.ts [hunk 25]
```diff
@@ -1,5 +1,6 @@
-import Onyx from 'react-native-onyx';
-import _ from 'underscore';
+import type {Mock} from 'jest-mock';
+import reactNativeOnyxMock from '../../__mocks__/react-native-onyx';
+// import Onyx from 'react-native-onyx';
```
Comment (fabioh8010) [inline tests/unit/NetworkTest.ts:3]: Same here about Onyx mock

File: tests/unit/NetworkTest.ts [hunk 26]
```diff
@@ -25,7 +28,7 @@ OnyxUpdateManager();
 const originalXHR = HttpUtils.xhr;
 
 beforeEach(() => {
-    global.fetch = TestHelper.getGlobalFetchMock();
+    global.fetch = TestHelper.getGlobalFetchMock() as typeof fetch;
```
Comment (fabioh8010) [inline tests/unit/NetworkTest.ts:31]: Same about `@ts-expect-error`

File: src/types/onyx/ReportAction.ts [hunk 27]
```diff
@@ -224,5 +226,7 @@ type ReportAction = ReportActionBase & OriginalMessage;
 
 type ReportActions = Record<string, ReportAction>;
 
+type ReportActionCollectionDataSet = CollectionDataSet<typeof ONYXKEYS.COLLECTION.REPORT_ACTIONS>;
```
[the remark this answers is not in the document]
Comment (pasyukevich) [reply src/types/onyx/ReportAction.ts:229]: Do we already have this helper?

File: tests/actions/ReportTest.ts [hunk 28]
```diff
@@ -530,7 +533,6 @@ describe('actions/Report', () => {
                         value: {
                             1: REPORT_ACTION,
                         },
-                        shouldNotify: true,
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:527]: WHy this change?

File: tests/unit/MigrationTest.ts [hunk 29]
```diff
@@ -0,0 +1,343 @@
...
+
+        it('Should skip if no valid reportActions', () => {
+            const setQueries = {
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1`]: null,
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}2`]: {},
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}3`]: {},
+                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}4`]: null,
+            };
+
+            // @ts-expect-error preset null values
```
Comment (VickyStash) [inline tests/unit/MigrationTest.ts:234]: If it's possible to set `null` values maybe we can update `CollectionDataSet` type to include null:
```
type CollectionDataSet<TCollectionKey extends OnyxCollectionKey> = Record<`${TCollectionKey}${string}`, OnyxEntry<OnyxCollectionValuesMapping[TCollectionKey]>>;
```
After that you can type `setQueries` var using `CollectionDataSet` type and get rid of this comment. What do you think?

File: tests/actions/ReportTest.ts (hunk 28 above)
Comment (pasyukevich) [reply tests/actions/ReportTest.ts:527]: reverted with the comment as it was done in another test

File: tests/unit/MigrationTest.ts (hunk 29 above)
Comment (pasyukevich) [reply tests/unit/MigrationTest.ts:234]: there is a problem not only with a type. 
In this case, we need to specify the key in onyx, since  `toCollectionDataSet` we can set it only from the object data by passing the selector (in this case we will select from the )

I think the current approach is simpler for test purposes

Comment (VickyStash) [reply tests/unit/MigrationTest.ts:234]: I mean if you update `CollectionDataSet` type as I mentioned above, you can then type it like

```
 const setQueries: CollectionDataSet<typeof ONYXKEYS.COLLECTION.REPORT_ACTIONS>  = {
                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}1`]: null,
                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}2`]: {},
                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}3`]: {},
                [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}4`]: null,
            };
```
No need to use `toCollectionDataSet` in this case, and you will not have TS error. You can even create a separate type for `CollectionDataSet<typeof ONYXKEYS.COLLECTION.REPORT_ACTIONS>` as you did [here](https://github.com/Expensify/App/pull/37482/files#diff-947a4c1dd5fa2e942610b43b2397e682a2fea075ea054a04dc2c6b812b5f5717R7).

But it's up to you

File: tests/actions/ReportTest.ts [hunk 30]
```diff
@@ -533,6 +533,8 @@ describe('actions/Report', () => {
                         value: {
                             1: REPORT_ACTION,
                         },
+                        // @ts-expect-error -- it's necessary for the test
+                        shouldNotify: true,
```
Comment (fabioh8010) [inline tests/actions/ReportTest.ts:537]: In order to get rid of this suppression (and in the other files too), you can change `emitOnyxUpdate` [here](https://github.com/Expensify/App/blob/main/tests/utils/PusherHelper.ts#L27) to use `OnyxServerUpdate`:

```ts
function emitOnyxUpdate(args: OnyxServerUpdate[]) {
```

**Review by fabioh8010 — APPROVED** [review]

**Review by aimane-chnaif — APPROVED** [review]

**Review by aimane-chnaif — APPROVED** [review]

**Review by bondydaa — APPROVED** [review]

**Comment by bondydaa** [comment]
lies, they had all passed.
